lib/ProductOpener/Import.pm

```diff
@@ -412,6 +412,167 @@ my %no = (
 );
 
 
+sub apply_organization_settings_to_product($$$$) {
+
+	my %stats = shift;
+	my $org_id = shift;
+	my $org_ref = shift;
+	my $args_ref = shift;
+
+	# Make sure the source as the authorization to load data to the org
+	# e.g. if an org has loaded fresh data manually or through Equadis,
+	# don't overwrite it with potentially stale CodeOnline or USDA data
+
+	if (defined $args_ref->{source_id}) {
+		if (not $org_ref->{"import_source_" . $args_ref->{source_id}}) {
+			$log->debug("skipping import for org without authorization for the source", { org_ref => $org_ref, source_id => $args_ref->{source_id} }) if $log->is_debug();
+			$stats{orgs_without_source_authorization}{$org_id} or $stats{orgs_without_source_authorization}{$org_id} = 0;
+			$stats{orgs_without_source_authorization}{$org_id}++;
+			next;
+		};
+	}
+
+	# The do_not_import_codeonline checkbox will be replaced by the new system above that will work for all sources
+
+	# Check if it is a CodeOnline import for an org with do_not_import_codeonline
+	if ((defined $args_ref->{source_id}) and ($args_ref->{source_id} eq "codeonline")
+		and (defined $org_ref->{do_not_import_codeonline}) and ($org_ref->{do_not_import_codeonline})) {
+		$log->debug("skipping codeonline import for org with do_not_import_codeonline", { org_ref => $org_ref }) if $log->is_debug();
+		next;
+	}
+	return;
+}
+
+
+
+sub get_org_id_for_product($$$$$$) {
+	
+	my $imported_product_ref = shift;
+	my $org_id = shift;
```

Comment:
if $org_id is passed as a parameter, if you change it inside the function, it won't change its value outside the function.

So instead, the function should return the $org_id, and it should be called like this:

my $org_id = get_org_id_for_product(...)